### ISSUE TITLE
Inconsistent address assignment in cluster.Rendezvous

### ISSUE THREAD

**Issue by rgngl** [asynkron/protoactor-go#701]
**Describe the bug**
[GetByClusterIdentity](https://github.com/asynkron/protoactor-go/blob/d89a24d6ef64ba401d4f3e4c123652f8d7524aa6/cluster/rendezvous.go#L31) and [GetByIdentity](https://github.com/asynkron/protoactor-go/blob/d89a24d6ef64ba401d4f3e4c123652f8d7524aa6/cluster/rendezvous.go#L68) methods in `cluster.Rendezvous` return different addresses for the same cluster identity under some conditions. When there are members providing different kinds in the same cluster, those two methods return different addresses.

**To Reproduce**
- Have a few instances of a cluster member that provide a certain grain kind, and one that doesn't.
- `GetByClusterIdentity` and `GetByIdentity` methods return different addresses for the same id.

**Expected behavior**
They should both return the same thing.

**Additional context**
This is creating a problem where we have a redis based lock for actors which is locked/unlocked when the actors are initialized and terminated.

Replacing the implementation of `GetByIdentity` with
```
func (r *Rendezvous) GetByIdentity(identity string) string {
	parts := strings.SplitN(identity, "/", 2)

	return r.GetByClusterIdentity(&ClusterIdentity{
		Kind:     parts[0],
		Identity: parts[1],
	})
}
``` 
seems to fix the problem but I'm not sure if it would break something else.

**Comment by rogeralsing** [comment]
Tests fail for the PR, we need to investigate this one more

**Comment by rgngl** [comment]
This is fixed by #775, so closing.
